server/artifacts/artifact_server.go

```diff
@@ -95,9 +95,12 @@ func (a *ArtifactServer) gateKeeping(r *http.Request) (context.Context, error) {
 	if token == "" {
 		cookie, err := r.Cookie("authorization")
 		if err != nil {
```

Comment:
this bug was that if there is no cookie (which there won't be in server auth-mode) we get an ErrNoCookie